concepts/eventing-code-guidelines.md

```diff
@@ -0,0 +1,1147 @@
+# Eventing team code guidelines
+
+**Document Intention** 
+
+The intention of this document is to provide style, testing, and code guidelines for components of the **eventing** team. Other teams in Kyma may follow different guidelines.
+The document is supposed to be a **living** document that can be changed by anyone in the team. However the team majority (50%) must agree on changes for this guideline. 
+
+This guide shall be publicly available so that it can be referenced when reviewing pull requests. The agreements of this guide shall not affect **external collaborators**. If an external PR does not align with our guidelines, we should accept the PR as it is and apply the guidelines ourselves in a follow-up PR so we don't block external contributors. Of course, also external PRs shall meet our quality standards but we shouldn't be too nitpicky.
+
+For now, the guidelines shall be applied to new code (PRs). There is no need to rewrite our entire code base now. This will gradually happen over time. As soon as the code reflects our guidelines, the need to look into this guide will decrease more and more.
+
+## Table of contents
+
+<!-- generate me with markdown-toc 
+```bash
+// source: https://github.com/jonschlinkert/markdown-toc
+markdown-toc -i --maxdepth 2 eventing-code-guidelines.md
+Do NOT TOUCH anything between the toc comments because this is used as a `marker` where to place the toc for markdown-toc.
+```
+-->
+
+<!-- toc -->
+
+<!-- tocstop -->
+
+## Recommended libraries
+
+The following section describes the **desired choice** of **libraries** for testing, logging, etc.
+
+### Testing
+
+**Testing frameworks**:
+
+Use [`t.Testing`](https://pkg.go.dev/testing) for unit tests and for controller integration tests.
+
+<!-- voting options for testing framework in controller integration tests:
+1. testing.T // proposed by guidelines
+1. [ginkgo](https://github.com/onsi/ginkgo)
+-->
+
+**Assertion libraries**:
+
+<!-- voting options for assertion libraries in unit tests:
+Right now we use both: testing.T and gomega. We should reach a consensus here to have consistency.
+
+1. testing.T
+2. [stretchr/testify](https://github.com/stretchr/testify)
+3. [onsi/gomega](https://github.com/onsi/gomega)
+-->
+
+Use [`gomega`](https://github.com/onsi/gomega) for writing test **assertions** in controller integration tests. Do not use [stretchr/testify](https://github.com/stretchr/testify) unless you have a reason for it.
```

Comment:
```suggestion
Use [`gomega`](https://github.com/onsi/gomega) for writing test **assertions** in controller integration tests. Do not use [stretchr/testify](https://github.com/stretchr/testify) unless you can justify it.
```

A reason could be "I felt like doing that for variety" but that's probably not a valid reason?